src/components/Releases/ReleaseArtifactsTab/utils/generic-key-value-rendering-utils.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+import {
+  DescriptionList,
+  DescriptionListGroup,
+  DescriptionListTerm,
+  DescriptionListDescription,
+} from '@patternfly/react-core';
+import { isObject } from 'lodash-es';
+import ExternalLink from '../../../../shared/components/links/ExternalLink';
+import { getImageLink, isUrl } from './url';
+
+/**
+ * converts a key (e.g. `build_id`, `commitHash`) into a more readable label
+ * by adding spaces and capitalizing only the first word.
+ *
+ * Examples:
+ * - "build_id"     -> "Build id"
+ * - "commitHash"   -> "Commit hash"
+ * - "last_updated" -> "Last updated"
+ */
+function humanizeKey(key: string): string {
+  const spaced = key
+    .replace(/_{2,}/g, '_') // Replace multiple underscores with a single underscore
+    .replace(/_/g, ' ') // Replace remaining underscores with spaces
+    .replace(/([a-z0-9])([A-Z])/g, '$1 $2'); // add space between camelCase words
+
+  return spaced.charAt(0).toUpperCase() + spaced.slice(1).toLocaleLowerCase(); // Capitalize only the first letter
+}
+
+export function renderKeyValueList(data: Record<string, unknown>) {
+  return (
+    <DescriptionList columnModifier={{ default: '2Col', md: '3Col' }}>
+      {Object.entries(data).map(([key, value]) => (
+        <DescriptionListGroup key={key}>
+          <DescriptionListTerm>{humanizeKey(key)}</DescriptionListTerm>
+          <DescriptionListDescription>{renderValue(value)}</DescriptionListDescription>
+        </DescriptionListGroup>
+      ))}
+    </DescriptionList>
+  );
+}
+
+function renderValue(value: unknown): React.ReactNode {
+  if (value === null || value === undefined) {
+    return <span>—</span>;
+  }
+
+  if (Array.isArray(value)) {
+    if (value.every((v) => typeof v === 'object' && v !== null && !Array.isArray(v))) {
+      // Array of objects
+      return (
+        <ul style={{ margin: 0 }}>
+          {value.map((obj, idx) => (
+            <li key={idx} style={{ marginBottom: '0.5rem' }}>
```

Comment:
do you have better method to set the margin?

Reply:
hmm, maybe we could use css, but I think it should be fine to leave it like this :)